src/components/Receiving/ReceivingList.js

```diff
@@ -180,6 +180,14 @@ class ReceivingList extends Component {
     this.setState({ isItemDetailsModalOpened: false });
   }
 
+  getItemDetails = () => {
+    const { itemDetails } = this.state;
+
+    Object.keys(itemDetails).forEach(key => !itemDetails[key].length && delete itemDetails[key]);
```

Comment:
`delete itemDetails[key]` - is direct mutating of state, it's against React philosophy. Let's discuss all aproach of passing data through modals, this shouldn't be so complicated.